### PR TITLE
Js 437 parallelize line emission after redshift bins

### PR DESCRIPTION
**Pull request by kenfus**

### CODE REVIEW

File: karabo/simulation/line_emission.py [hunk 1]
```diff
@@ -489,41 +598,66 @@ def line_emission_pointing(
...
+                beam_type,
+                gaussian_fwhm,
+                gaussian_ref_freq,
+                start_time,
+                obs_length,
+                cut,
+                img_size,
+                circle,
+                rascil,
+                verbose,
```
Comment (Lukas113) [inline karabo/simulation/line_emission.py:653]: Not sure if it's best practice or not. But I personally prefer using keyword assignment in the functions of your own repository or other dependencies which are under development. The reason is simply maintenance. If an additional argument is introduced in between with a default value, everything breaks. This is not the case with key-word assignment.

**Review by Lukas113 — CHANGES_REQUESTED** [review]

File: karabo/simulation/line_emission.py [hunk 2]
```diff
@@ -207,15 +214,17 @@ def plot_scatter_recon(
...
+    )
     plt.colorbar(recon_img, ax=ax2, label="Flux Density [Jy]")
 
     plt.tight_layout()
     plt.savefig(outfile + ".pdf")
 
 
 def sky_slice(
-    sky: SkyModel, z_obs: NDArray[np.float_], z_min: float, z_max: float
+    sky: SkyModel, z_obs: NDArray[np.float_], z_min: np.float_, z_max: np.float_
```
[the remark this answers is not in the document]
Comment (jejestern) [reply karabo/simulation/line_emission.py:227]: No, we cannot remove the #TODO. I need to do some additional changes in a next step to adjust to the changes from merge #400.

[the remark this answers is not in the document]
Comment (jejestern) [reply karabo/simulation/line_emission.py:227]: The z_obs will never be an int. This is a simulated dataset with observed redshifts and therefore int makes no sense at all. We can however adjust to a general float type if this exists.

[the remark this answers is not in the document]
Comment (Lukas113) [reply karabo/simulation/line_emission.py:227]: If it's never an integer then it's fine. But then as you suggested, I think a Union[float, np.float_] whould make sense. I don't see such a dtype in _types.py yet, so feel free to introduce it there.

**Review by Lukas113 — APPROVED** [review]

**Pull request merged**